examples/vesting-wallet/src/constructor.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.21;
+
+contract VestingWalletExample {
+    address private _owner;
+
+    uint256 private _released;
+    mapping(address token => uint256) private _erc20Released;
+    uint64 private _start;
+    uint64 private _duration;
+
+    error OwnableInvalidOwner(address owner);
+    event OwnershipTransferred(
+        address indexed previousOwner,
+        address indexed newOwner
+    );
+
+    constructor(address beneficiary, uint64 startTimestamp, uint64 durationSeconds) payable {
+         if (beneficiary == address(0)) {
+            revert OwnableInvalidOwner(address(0));
+        }
+        _transferOwnership(beneficiary);
```

Comment:
Nit: it would be fine to just set `beneficiary` like `_owner = beneficiary`, even without emitting an event. The reason is that we do not care here about `Ownable` but about `VestingWallet`.